app/javascript/mastodon/components/status_content.js

```diff
@@ -46,10 +51,15 @@ export default class StatusContent extends React.PureComponent {
     }
   }
 
+  componentDidMount () {
+    this._updateStatusLinks();
+  }
+
   componentDidUpdate () {
     if (this.props.onHeightUpdate) {
       this.props.onHeightUpdate();
     }
+    this._updateStatusLinks();
```

Comment:
You can use `prevProps` argument to detect status changes, instead of `className` way.

```js
  componentDidUpdate (prevProps) {
    if (this.props.onHeightUpdate) {
      this.props.onHeightUpdate();
    }
    if (prevProps.status !== this.props.status) {
      this._updateStatusLinks();
    }
  }
```